executor/builder.go

```diff
@@ -1420,6 +1422,12 @@ func (b *executorBuilder) buildProjection(v *plannercore.PhysicalProjection) Exe
 	if int64(v.StatsCount()) < int64(b.ctx.GetSessionVars().MaxChunkSize) {
 		e.numWorkers = 0
 	}
+
+	// Use un-parallel projection for write query to avoid concurrent on memdb.
+	// See also https://github.com/pingcap/tidb/issues/26832
+	if b.inUpdateStmt || b.inDeleteStmt || b.inInsertStmt {
```

Comment:
Maybe we also need to handle `SELECT FOR UPDATE`

Reply:
Done @coocood